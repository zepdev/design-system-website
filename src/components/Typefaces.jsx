import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import LaunchIcon from './icons/LaunchIcon'
import theme from '../data/theme'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginBottom: theme.spacing.xl * 2,
  },
  container: {
    flexBasis: '48%',
  },
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 136,
    background: theme.colors.gray.grayLighter.hex,
  },
  title: {
    fontSize: 28,
  },
  buttonContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    background: theme.colors.gray.grayLightest.hex,
    paddingTop: theme.spacing.m,
    paddingBottom: theme.spacing.m,
    paddingLeft: theme.spacing.l,
    paddingRight: theme.spacing.l,
    borderTop: `1px solid ${ theme.colors.gray.grayLight.hex }`,
  },
  text: {
    fontSize: 12,
    textTransform: 'capitalize',
  },
  link: {
    color: 'inherit',
    '&:hover': {
      color: theme.colors.primary.indigoBlue.hex,
    },
  },
})

const Typefaces = ({ classes }) => (
  <div className={classes.root}>
    {Object.keys(theme.typography.typefaces).map((elem, idx) => (
      <div className={classes.container} key={`typeface${ idx }`}>
        <div className={classes.titleContainer}>
          <p className={classes.title} style={{ fontFamily: elem }}>
            Lorem Ipsum
          </p>
        </div>
        <div className={classes.buttonContainer}>
          <p className={classes.text}>{elem} Font Family</p>
          <a
            href={`https://fonts.google.com/specimen/${ elem.charAt(0).toUpperCase() +
              elem.slice(1).toLowerCase() }`}
            target="_blank"
            rel="noopener"
            aria-label="Launch"
            className={classes.link}
          >
            <LaunchIcon width="24" height="24" />
          </a>
        </div>
      </div>
    ))}
  </div>
)

Typefaces.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Typefaces)
