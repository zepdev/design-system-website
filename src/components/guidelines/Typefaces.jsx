import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import OpenInNewIcon from '../icons/OpenInNewIcon'
import theme from 'zeppelin-element-library/bundle/themes/theme.json'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginBottom: `${ theme.spacing.component.xxl.rem }rem`,
  },
  container: {
    flexBasis: '100%',
    paddingBottom: `${ theme.spacing.component.m.rem }rem`,
  },
  [`@media (min-width: ${ theme.breakpoints.s })`]: {
    container: {
      flexBasis: '48%',
      paddingBottom: 0,
    },
  },
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 136,
    background: theme.colors.gray.grayLighter.hex,
  },
  title: {
    fontSize: 28,
  },
  buttonContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    background: theme.colors.gray.grayLightest.hex,
    paddingTop: theme.spacing.component.s.px,
    paddingBottom: theme.spacing.component.s.px,
    paddingLeft: theme.spacing.component.l.px,
    paddingRight: theme.spacing.component.l.px,
    borderTop: `1px solid ${ theme.colors.gray.grayLight.hex }`,
  },
  text: {
    fontSize: 12,
    textTransform: 'capitalize',
  },
  link: {
    color: 'inherit',
    '&:hover': {
      color: theme.colors.primary.indigoBlue.hex,
    },
  },
})

const Typefaces = ({ classes }) => (
  <div className={classes.root}>
    {Object.keys(theme.typography.typefaces).map((elem, idx) => (
      <div className={classes.container} key={`typeface${ idx }`}>
        <div className={classes.titleContainer}>
          <p className={classes.title} style={{ fontFamily: elem }}>
            Lorem Ipsum
          </p>
        </div>
        <div className={classes.buttonContainer}>
          <p className={classes.text}>{elem} Font Family</p>
          <a
            href={`https://fonts.google.com/specimen/${ elem.charAt(0).toUpperCase() +
              elem.slice(1).toLowerCase() }`}
            target="_blank"
            rel="noopener"
            aria-label="Launch"
            className={classes.link}
          >
            <OpenInNewIcon width="24" height="24" />
          </a>
        </div>
      </div>
    ))}
  </div>
)

Typefaces.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Typefaces)
