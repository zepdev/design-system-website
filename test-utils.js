import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
