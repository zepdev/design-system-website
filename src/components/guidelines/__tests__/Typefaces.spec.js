import React from 'react'
import { render } from '../../../../test-utils'
import Typefaces from '../Typefaces'

describe('Typefaces', () => {
  it('renders correctly', () => {
    const component = render(<Typefaces />)
    expect(component).toMatchSnapshot()
  })
})
