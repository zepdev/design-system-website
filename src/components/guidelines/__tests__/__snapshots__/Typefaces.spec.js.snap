// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typefaces renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="makeStyles-container-2"
        >
          <div
            class="makeStyles-titleContainer-3"
          >
            <p
              class="makeStyles-title-4"
              style="font-family: Roboto, sans-serif;"
            >
              Lorem Ipsum
            </p>
          </div>
          <div
            class="makeStyles-buttonContainer-5"
          >
            <p
              class="makeStyles-text-6"
            >
              roboto
               Font Family
            </p>
            <a
              aria-label="Launch"
              class="makeStyles-link-7"
              href="https://fonts.google.com/specimen/Roboto"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="title_launch"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_launch"
                >
                  Open In New
                </title>
                <path
                  d="M14,3 L14,5 L17.59,5 L7.76,14.83 L9.17,16.24 L19,6.41 L19,10 L21,10 L21,3 L14,3 Z M19,19 L5,19 L5,5 L12,5 L12,3 L5,3 C3.89,3 3,3.9 3,5 L3,19 C3,20.1045695 3.8954305,21 5,21 L19,21 C20.1045695,21 21,20.1045695 21,19 L21,12 L19,12 L19,19 Z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="makeStyles-container-2"
        >
          <div
            class="makeStyles-titleContainer-3"
          >
            <p
              class="makeStyles-title-4"
              style="font-family: Roboto Condensed, sans-serif;"
            >
              Lorem Ipsum
            </p>
          </div>
          <div
            class="makeStyles-buttonContainer-5"
          >
            <p
              class="makeStyles-text-6"
            >
              robotoCondensed
               Font Family
            </p>
            <a
              aria-label="Launch"
              class="makeStyles-link-7"
              href="https://fonts.google.com/specimen/Roboto+Condensed"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="title_launch"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_launch"
                >
                  Open In New
                </title>
                <path
                  d="M14,3 L14,5 L17.59,5 L7.76,14.83 L9.17,16.24 L19,6.41 L19,10 L21,10 L21,3 L14,3 Z M19,19 L5,19 L5,5 L12,5 L12,3 L5,3 C3.89,3 3,3.9 3,5 L3,19 C3,20.1045695 3.8954305,21 5,21 L19,21 C20.1045695,21 21,20.1045695 21,19 L21,12 L19,12 L19,19 Z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="makeStyles-container-2"
      >
        <div
          class="makeStyles-titleContainer-3"
        >
          <p
            class="makeStyles-title-4"
            style="font-family: Roboto, sans-serif;"
          >
            Lorem Ipsum
          </p>
        </div>
        <div
          class="makeStyles-buttonContainer-5"
        >
          <p
            class="makeStyles-text-6"
          >
            roboto
             Font Family
          </p>
          <a
            aria-label="Launch"
            class="makeStyles-link-7"
            href="https://fonts.google.com/specimen/Roboto"
            rel="noopener"
            target="_blank"
          >
            <svg
              aria-labelledby="title_launch"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_launch"
              >
                Open In New
              </title>
              <path
                d="M14,3 L14,5 L17.59,5 L7.76,14.83 L9.17,16.24 L19,6.41 L19,10 L21,10 L21,3 L14,3 Z M19,19 L5,19 L5,5 L12,5 L12,3 L5,3 C3.89,3 3,3.9 3,5 L3,19 C3,20.1045695 3.8954305,21 5,21 L19,21 C20.1045695,21 21,20.1045695 21,19 L21,12 L19,12 L19,19 Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="makeStyles-container-2"
      >
        <div
          class="makeStyles-titleContainer-3"
        >
          <p
            class="makeStyles-title-4"
            style="font-family: Roboto Condensed, sans-serif;"
          >
            Lorem Ipsum
          </p>
        </div>
        <div
          class="makeStyles-buttonContainer-5"
        >
          <p
            class="makeStyles-text-6"
          >
            robotoCondensed
             Font Family
          </p>
          <a
            aria-label="Launch"
            class="makeStyles-link-7"
            href="https://fonts.google.com/specimen/Roboto+Condensed"
            rel="noopener"
            target="_blank"
          >
            <svg
              aria-labelledby="title_launch"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_launch"
              >
                Open In New
              </title>
              <path
                d="M14,3 L14,5 L17.59,5 L7.76,14.83 L9.17,16.24 L19,6.41 L19,10 L21,10 L21,3 L14,3 Z M19,19 L5,19 L5,5 L12,5 L12,3 L5,3 C3.89,3 3,3.9 3,5 L3,19 C3,20.1045695 3.8954305,21 5,21 L19,21 C20.1045695,21 21,20.1045695 21,19 L21,12 L19,12 L19,19 Z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
